meta-balena-common/recipes-kernel/linux-firmware/linux-firmware_%.bbappend

```diff
@@ -1,3 +1,43 @@
+# Cleanup iwlwifi firmware files
+IWLWIFI_PATH = "lib/firmware"
+IWLWIFI_REGEX = "^iwlwifi-([0-9a-zA-Z-]+)-([0-9]+).ucode$"
+
+IWLWIFI_FW_MIN_API[7260] = "17"
+IWLWIFI_FW_MIN_API[7265] = "17"
+IWLWIFI_FW_MIN_API[7265D] = "29"
+IWLWIFI_FW_MIN_API[8000C] = "34"
+IWLWIFI_FW_MIN_API[8265] = "34"
+
+python() {
+    import os,re
+
+    workdir = d.getVar('WORKDIR', True)
+    destination = d.getVar('D', True)
+    if destination is None:
+        return
+    install_path = os.path.join(destination, d.getVar('IWLWIFI_PATH',True))
+    package_name = d.getVar('PN', True)
+    regex = d.getVar('IWLWIFI_REGEX', True)
+    minapi_all = d.getVarFlags('IWLWIFI_FW_MIN_API') or {}
+    for chipset, minapi in minapi_all.items():
+        minapi = int(minapi_all[chipset] or 0)
+        bb.note('Limiting iwlwifi firmware for chipset {} to minimum API version {}.'.format(chipset, minapi))
+        package_files = []
+        extracted_path = os.path.join(workdir, '{}-{}'.format(package_name, d.getVar('PV', True)))
+        if not os.path.exists(extracted_path):
+            return
```

Comment:
This segment can be moved outside of the `for` statement, since it does not depend on the contents of `miniapi_all`, and also this way the `extracted_path` will be populated only once.